x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/runtimefields/RuntimeFieldsFeatureSetUsage.java

```diff
@@ -0,0 +1,272 @@
+                if (runtimeObject instanceof Map) {
+                    @SuppressWarnings("unchecked")
+                    Map<String, Object> runtimeMappings = (Map<String, Object>) runtimeObject;
+                    for (Object runtimeFieldMappingObject : runtimeMappings.values()) {
+                        if (runtimeFieldMappingObject instanceof Map) {
+                            @SuppressWarnings("unchecked")
+                            Map<String, Object> runtimeFieldMapping = (Map<String, Object>) runtimeFieldMappingObject;
+                            Object typeObject = runtimeFieldMapping.get("type");
+                            if (typeObject != null) {
+                                String type = typeObject.toString();
+                                RuntimeFieldStats stats = fieldTypes.computeIfAbsent(type, RuntimeFieldStats::new);
+                                stats.count++;
+                                if (indexFieldTypes.add(type)) {
+                                    stats.indexCount++;
+                                }
+                                Object scriptObject = runtimeFieldMapping.get("script");
+                                if (scriptObject == null) {
+                                    stats.scriptLessCount++;
+                                } else if (scriptObject instanceof Map) {
+                                    @SuppressWarnings("unchecked")
+                                    Map<String, Object> script = (Map<String, Object>) scriptObject;
+                                    Object sourceObject = script.get("source");
+                                    if (sourceObject != null) {
+                                        String scriptSource = sourceObject.toString();
+                                        int chars = scriptSource.length();
+                                        long lines = scriptSource.lines().count();
+                                        int docUsages = countOccurrences(scriptSource, "doc[\\[\\.]");
+                                        int sourceUsages = countOccurrences(scriptSource, "params\\._source");
+                                        stats.update(chars, lines, sourceUsages, docUsages);
+                                    }
+                                    Object langObject = script.get("lang");
+                                    if (langObject != null) {
+                                        stats.scriptLangs.add(langObject.toString());
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        RuntimeFieldStats[] runtimeFieldStats = fieldTypes.values().toArray(new RuntimeFieldStats[0]);
+        Arrays.sort(runtimeFieldStats, Comparator.comparing(RuntimeFieldStats::type));
+        return new RuntimeFieldsFeatureSetUsage(runtimeFieldStats);
```

Comment:
Any reason not to use a `List` in the response? I know we sometimes use `List` and sometimes arrays but I've never been sure why.

Reply:
I generally use a list when elements need to be added, and an array otherwise. I guess you'd like to use a list because it can be made truly immutable?

Reply:
Mostly because we accumulated a list so it feels like we may as well keep it that way. Either way is fine.